### DISCUSSION TITLE
How does the Updated timestamp work?

### DISCUSSION THREAD

**Discussion by pmatulis** [mmistakes/minimal-mistakes#3049] [Q&A] [answered]
1. What brings in (enables) the **Updated** timestamp feature for posts?

1. How does this timestamp get updated? I assumed that editing a post would automatically trigger it but that doesn't work for me. And when I change the `date:` key of a post the page cannot be found anymore. I guess because it no longer corresponds to the filename?

1. Finally, is it possible to move the timestamp to the top of a post? The default is to place it at the bottom.

**Comment by mmistakes** [comment, marked as answer]
You need to add some YAML front matter for the updated timestamp. This works on post's which automatically have a date based off of the filename you give it or if you add something like this to your front matter:

```yaml
---
date: 2021-07-07
---
```

To get the updated value you need to add `last_modified_at` to front matter:

```yaml
---
last_modified_at: 2021-07-07
---
```

Many of the core Jekyll plugins utilize `last_modified_at` timestamps which is why I choose that variable name. There is a [Jekyll plugin](https://github.com/gjtorikian/jekyll-last-modified-at) that automatically adds that value based off of the last time the file was modified if you don't want to manually at `last_modified_at` datetimes to your posts and pages.

**Reply by mmistakes** [reply]
And for **2**, `date` front matter overrides whatever you set as the filename. Which is why "you can't find" the post anymore... depending on what your permalink structure is it the URL probably changed.

So if you take a file named `2020-01-01-my-post.md` it will have a `date` of `2020-01-01`.
But if you add some front matter like `date: 2021-02-02`, this will override the preceding date and you'll get `2021-02-02` instead.

Which if you're using the default Jekyll permalink pattern will give you a URL of `_site/2021/02/02/my-post.html`.

**Reply by pmatulis** [reply]
Thank you for answering. And thank you for your theme. :+1:
